src/python/pants/backend/python/util_rules/lockfile.py

```diff
@@ -0,0 +1,179 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+"""
+Helpers for tools to generate their lockfiles
+TODO:
+- TODO: helpful error messages on Tools missing required fields
+"""
+
+from __future__ import annotations
+
+from typing import Iterable, Type
+
+from typing_extensions import Protocol
+
+from pants.backend.python.goals import lockfile
+from pants.backend.python.goals.lockfile import GeneratePythonLockfile
+from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.backend.python.subsystems.setup import PythonSetup
+from pants.backend.python.target_types import InterpreterConstraintsField
+from pants.backend.python.util_rules.partition import _find_all_unique_interpreter_constraints
+from pants.core.goals.generate_lockfiles import GenerateToolLockfileSentinel
+from pants.engine.rules import rule
+from pants.engine.target import FieldSet
+from pants.engine.unions import UnionRule
+from pants.util.logging import LogLevel
+from pants.util.memo import memoized
+from pants.util.ordered_set import FrozenOrderedSet
+from pants.util.strutil import softwrap
+
+
+class LockfileRules:
+    """The type of lockfile generation strategy to use for a tool."""
+
+    @staticmethod
+    def custom(cls: Type[PythonToolBase]) -> Iterable:
+        """The plugin author defines its own export machinery."""
+        yield from lockfile.rules()
+
+    @staticmethod
+    def from_tool(cls: Type[PythonToolBase]) -> Iterable:
+        """The tool can be just pip-installed."""
```

Comment:
Hmmmm all these tools can be `pip`-installed. Maybe add on a qualifier or two? "... with no other complications"?

Reply:
I guess the "just" falls more into filler word, how's "The tool can be installed with pip simply with `pip install mytool`."?